src/illumidesk/spawners/hooks.py

```diff
@@ -0,0 +1,39 @@
+import os
+import shutil
+from jupyterhub.spawner import Spawner
+
+
+def custom_auth_state_hook(spawner: Spawner, auth_state: dict) -> None:
+    """
+    Customized hook to assign USER_ROLE environment variable to LTI user role.
+    The USER_ROLE environment variable is used to select the notebook image based
+    on the user's role.
+    """
```

Comment:
I think it would be somewhat clear that this method should be assigned to the `Spawner.auth_state_hook` config, but we should document that explicitly here too.